server/cronjob/monitor.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package cronjob
+
+import (
+	"context"
+	"time"
+
+	"github.com/iotexproject/iotex-core/blockchain"
+	"github.com/iotexproject/iotex-core/p2p"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+// CronJob interface
+type CronJob interface {
+	Run()
+	Interval() time.Duration
+}
+
```

Comment:
we can use recurring task to do this, search `NewRecurringTask(` in the codebase you will see